### PR TITLE
[RFR] Backward compatibility

### PR DESCRIPTION
**Pull request by jpetitcolas**
- [x] Fix build process (missing configuration part)
- [x] Add BC proxy methods to ConfigurationProvider
- [x] Test it on real-world project
- [x] Update README and UPGRADE files

### CODE REVIEW

**Comment by fzaninotto** [comment]
Don't forget the README and UPGRADE files

File: src/javascripts/ng-admin/es6/lib/Factory.js [hunk 1]
```diff
@@ -46,6 +46,10 @@ class Factory {
         this._fieldTypes[name] = constructor;
     }
 
+    getFieldConstructor(name) {
```
Comment (jpetitcolas) [inline src/javascripts/ng-admin/es6/lib/Factory.js]: Added to ease inheritance for custom types.

**Comment by jpetitcolas** [comment]
Switching to RFR. Tests pass locally.

File: UPGRADE-0.7.md [hunk 2]
```diff
@@ -6,17 +6,34 @@ ng-admin 0.7 breaks compatibility with 0.5 and makes the "factories" configurati
 
 ## Configuration factory
 
-The configuration is now done through the `AdminDescription` object. You can retrieve it directly from Angular DI. You just have to retrieve your application from this factory instead of the `NgAdminConfigurationProvider` as previously. In addition, `NgAdminConfigurationProvider` was renamed to `NgAdminProvider`:
+The configuration is now done through the `AdminDescription` object. You can retrieve it directly from Angular DI.
 
 ``` diff
 - app.config(function (NgAdminConfigurationProvider, RestangularProvider) {
-+ app.config(function (AdminDescription, NgAdminProvider, RestangularProvider) {
++ app.config(function (AdminDescription, NgAdminConfigurationProvider, RestangularProvider) {
```
Comment (fzaninotto) [inline UPGRADE-0.7.md]: remove AdminDescription

File: UPGRADE-0.7.md [hunk 3]
```diff
@@ -6,17 +6,34 @@ ng-admin 0.7 breaks compatibility with 0.5 and makes the "factories" configurati
...
 
 -     nga.configure(admin);
-+     NgAdminProvider.configure(admin);
++     NgAdminConfigurationProvider.configure(admin);
   });
 ```
+
+Note you still need the `NgAdminConfigurationProvider` to apply your application configuration.
+
+During the 0.8 version, some proxy methods transmits actions from `NgAdminConfigurationProvider` to `AdminDescription`, allowing you to transition smoothly.
```
Comment (fzaninotto) [inline UPGRADE-0.7.md]: please remove

File: UPGRADE-0.7.md [hunk 4]
```diff
@@ -6,17 +6,34 @@ ng-admin 0.7 breaks compatibility with 0.5 and makes the "factories" configurati
 
 ## Configuration factory
```
Comment (fzaninotto) [inline UPGRADE-0.7.md]: You can remove the entire paragraph.

File: README.md [hunk 5]
```diff
@@ -80,7 +80,7 @@ Here is a full example for a backend that will let you create, update, and delet
 
 var app = angular.module('myApp', ['ng-admin']);
 
-app.config(function (AdminDescription, NgAdminProvider) {
+app.config(function (AdminDescription, NgAdminConfigurationProvider) {
```
Comment (fzaninotto) [inline README.md]: remove AdminDescription

File: README.md [hunk 6]
```diff
@@ -156,7 +156,7 @@ app.config(function (AdminDescription, NgAdminProvider) {
                 .template('<other-page-link></other-link-link>')
         ]);
 
-    NgAdminProvider.configure(app);
+    NgAdminConfigurationProvider.configure(app);
```
Comment (fzaninotto) [inline README.md]: nga

File: README.md [hunk 7]
```diff
@@ -261,7 +261,7 @@ Defines the API endpoint for a view. It can be a string or a function.
 
 The `dashboardView` also defines `sortField` and `sortDir` fields like the `listView`.
 
-* `limit(Number)`
+* `.perPage(Number)`
```
Comment (fzaninotto) [inline README.md]: leave the leading dot

**Comment by jpetitcolas** [comment]
Files updated. But tests are still failing on Travis only. Should we merge it?

**Pull request merged**